extensions-core/kafka-indexing-service/src/main/java/org/apache/druid/indexing/kafka/supervisor/KafkaSupervisor.java

```diff
@@ -293,6 +293,30 @@ protected Map<Integer, Long> getRecordLagPerPartition(Map<Integer, Long> current
         );
   }
 
+  @Override
+  // suppress use of CollectionUtils.mapValues() since the valueMapper function is dependent on map key here
+  @SuppressWarnings("SSBasedInspection")
+  // Used while generating Supervisor lag reports per task
+  protected Map<Integer, Long> getRecordLagPerPartition(Map<Integer, Long> currentOffsets)
+  {
+    if (latestSequenceFromStream == null || currentOffsets == null) {
+      return Collections.emptyMap();
+    }
+
+    return currentOffsets
+        .entrySet()
+        .stream()
+        .filter(e -> latestSequenceFromStream.get(e.getKey()) != null)
```

Comment:
The old method reports the lag for every partition present in `latestSequenceFromStream`.
But this new one also seems to be doing the same thing because you are filtering the entries using the `latestSequenceFromStream`.

I am perhaps missing something but not sure how the two methods are different.
Please also add a test to verify the difference between the two results.

Reply:
Also will add a unit test for the reports.